### DISCUSSION TITLE
Route Post Form Arrays

### DISCUSSION THREAD

**Discussion by pietru2004** [pocketbase/pocketbase#3638] [Q&A] [answered]
I have read that if I name input field MyArray[text] it will make json array called myarray with key text and value of input in PHP...
Is there way to do that with pocketbase js extend? I mean to name field in specific way and get Array for JSON fields in db?

Related: https://stackoverflow.com/questions/9073690/post-an-array-from-an-html-form-without-javascript

**Comment by ganigeorgiev** [comment]
You can submit array values by duplicating the input:
```html
<!-- on the server-side this should be received as ["value1", "value2"] -->
<input name="yourField" value="value1"/>
<input name="yourField" value="value2"/>
```
But there is no map autovivification in PocketBase (_at least not out of the box_).

**Reply by pietru2004** [reply]
if my update code is 
```js
update_record: (c) => {
        let id = c.pathParam("id");
        id = id.replace(/'/g, "");
        try {
            const record = $app.dao().findRecordById("blocks",id)
        
            const form = new RecordUpsertForm($app, record);
        
            form.loadRequest(c.request(), "")
            form.submit()
        } catch (err) {
            console.log(err)
            return c.redirect(400, "/error");
        }

        return 
    },
```

can I also use 
```js
form.loadData({
    "title":          "Lorem ipsum",
    "active":         true,
    "someOtherField": 123,
})
```

idea is I have collection in pocketbase and it has 2 numner fields and 1 json field called content, and I want to each incoming keys prefixed as `content_` (or `content[value_name]`) to be in content json field

**Reply by ganigeorgiev** [reply, marked as answer]
You can use both `loadRequest` and `loadData` if that is what you are asking, but the field prefix/name parsing you'll have to perform your own. You can access the request body params as map via the `const data = $apis.requestInfo(c).data` helper (_must be called somewhere before the `c.request()` read call_).

_Side-note: you don't need to sanitize the id before passing it to `findRecordById()`, it is safe to be submitted as it is since the call uses named placeholder parameters for the query under the hood._

**Reply by pietru2004** [reply]
ok thanks for help

**Reply by pietru2004** [reply]
So I have tried this
```js
try {
            const record = $app.dao().findRecordById("blocks",id)
        
            const form = new RecordUpsertForm($app, record);
            
            const data = $apis.requestInfo(c).data
            let content = {}
            Object.entries(data).forEach(([key, value]) => {
                if (key.startsWith("content_")){
                    content[key.replace("content_","")]=value
                }
            });
        
            form.loadRequest(c.request(), "")
            form.loadData({
                "content":content
            })
            form.submit()
        } catch (err) {
            console.log(err)
            //return c.redirect(400, "/error");
        }

        return 
    },
```
But it gives me `GoError: Unsupported request content-type`
Also I tried doing `JSON.stringify()` on the content dictionary

**Reply by ganigeorgiev** [reply]
The "unsupported request content-type" is probably from the `loadRequest` call.
Currently only `application/json` and `multipart/form-data` requests are supported.
If you are submitting the request with a plain HTML form, you can add `enctype="multipart/form-data"` to the form tag.

**Reply by pietru2004** [reply]
erm yep seams I forgot to add that...
